service/resource/certificate/update_test.go

```diff
@@ -439,6 +460,10 @@ func Test_Resource_Certificate_ProcessUpdateState(t *testing.T) {
 			t.Fatalf("%d: error returned reading directory: %s\n", index, err)
 		}
 
+		if len(fileInfos) == 0 && len(test.expectedCertificateFiles) > 0 {
```

Comment:
This improves the testing, in the situation where there are no files written, but the test expects files to be there.